pkg/cmd/pr/shared/preserve_input.go

```diff
@@ -0,0 +1,52 @@
+package shared
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/cli/cli/pkg/iostreams"
+)
+
+func PreserveInput(io *iostreams.IOStreams, ims *IssueMetadataState, defs Defaults, doPreserve *bool) func() {
+	return func() {
+		if ims.Body == defs.Body && ims.Title == defs.Title {
+			return
+		}
+
+		if !*doPreserve {
+			return
+		}
+
+		out := io.ErrOut
+
+		// this extra newline guards against appending to the end of a survey line
+		fmt.Fprintln(out)
+
+		data, err := json.Marshal(ims)
```

Comment:
I like writing the output in json it allows flexibility for if we want to add in recovery options later on but is still usable to end users even without automatic recovery options using `jq`.